### ISSUE TITLE
AAD Authentication not working in 

### ISSUE THREAD

**Issue by ckcampbell248** [Microsoft/AzureStorageExplorer#56]
**Storage Explorer Version:**
0.9.6 Build 20180223.4
**OS Version:**
Windows 10 Enterprise
**Steps to Reproduce:**

1. Attempt to log in to Azure AD
2. Choose PIN or SmartCard or UserID and Password. 
3. Validate 2nd factor

**Expected Experience:**
Connect to my Azure account and see all of my storage accounts. 

**Actual Experience:**
The spinning "Waiting for Authorization" dialog disappears as if authentication succeeded but I don't get connected to my accounts. Also, if I use Azure Authenticator as the second factor, it works (says "Approved). In addition, no matter what authentication type I choose, I receive a text message with a code...even if I use Pin or SmartCard or Azure Authenticator. 

This was working prior to a couple of weeks ago. I just updated to the new version hoping to resolve this issue. I can connect to a storage account with a key just fine. 

Other Azure AD authenticated apps work fine.

**Comment by jinglouMSFT** [comment]
Can you try to download version 1.0.0 and give this a try and let us know?  You can download it from: https://azure.microsoft.com/en-us/features/storage-explorer/
Thanks.

**Comment by MRayermannMSFT** [comment]
@afyles , when you say accounts, do you mean subscriptions or user accounts?

**Comment by afyles** [comment]
I have an organizational account that is linked to several subscriptions.  I can post the log tomorrow.

**Comment by afyles** [comment]
![image](https://user-images.githubusercontent.com/354023/39491675-24b12fe0-4d5b-11e8-859c-82eee6716239.png)

![image](https://user-images.githubusercontent.com/354023/39491695-2f31ff80-4d5b-11e8-96b6-2f095263f19c.png)

**Comment by MRayermannMSFT** [comment]
And reentering your credentials does not fix it, correct? (potentially stupid question, I know, just need to make sure 😃).

**Comment by afyles** [comment]
Re-enter does not work, remove and re-add doesn't work.  I tried running as an administrator, pretty much everything.

**Comment by mmclane** [comment]
I think I might be seeing similar issues with 1.1 and macOS.

**Comment by MRayermannMSFT** [comment]
@mmclane , can you describe exactly what issues you are seeing?

**Comment by mmclane** [comment]
Yep, https://github.com/Microsoft/AzureStorageExplorer/issues/259
